File: SHM/Pini.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <sys/ipc.h>
#include <sys/shm.h>
#include <semaphore.h>
#include <fcntl.h>
#include <unistd.h>

#define SHM_SIZE 1024

int main() {
    int shm_id;
    char *shm_ptr;
    key_t key = ftok("shmfile", 65);

    shm_id = shmget(key, SHM_SIZE, 0666 | IPC_CREAT);
    shm_ptr = (char *)shmat(shm_id, NULL, 0);

    sprintf(shm_ptr, "003INIT");
    printf("Memoria compartida inicializada: %s\n", shm_ptr);

    shmdt(shm_ptr);

    return 0;
}

File: SHM.py/Pini.py
import mmap
import os
import posix_ipc

SHM_NAME = "/shmfile"

shm = posix_ipc.SharedMemory(SHM_NAME, posix_ipc.O_CREX, size=1024)
memory = mmap.mmap(shm.fd, shm.size)
os.close(shm.fd)

memory.write(b"003INIT")
print("Memoria compartida inicializada: 003INIT")

memory.close()
